src/main/java/net/minecraftforge/common/DimensionManager.java

```diff
@@ -408,6 +457,31 @@ public static void readRegistry(CompoundNBT data)
                 registerDimensionInternal(entry.id, entry.name, mod, entry.data == null ? null : new PacketBuffer(Unpooled.wrappedBuffer(entry.data)), entry.skyLight());
             }
         }
+
+        foldersScheduledForDeletion.clear();
+
+        ListNBT folderList = data.getList("delete_dirs", Constants.NBT.TAG_STRING);
+        folderList.stream()
+                .map(INBT::getString)
+                .forEach(foldersScheduledForDeletion::add);
+    }
+
+    public static void processScheduledDeletions(SaveHandler saveHandler)
```

Comment:
Is there a way we can add a confirmation screen similiar to the registry issues one?
I don't like the idea of deleting files in bulk without atleast a confirmation.
Also, should filter this list with a folder.exists() so we don't warn we're deleting folders that are already gone.

Reply:
If that gets added then i'd want an option to opt out. The dimensions in my mod are supposed to be treated as scrappable timed items, not like registries. Would be very annoying to have that dialog every time you start the game again. 

I specifically warn players not to build in those dimensions and that they are timed.

Reply:
Then your mod setup is bad, Dimensions are large objects filled with user data. And should not be treated as lightweight trash objects.
You should rethink your design and see if its possible to get away with reusing the dimensions and just resetting the changes users do.

Reply:
I didn't mean treated as lightweight, i meant as something that shouldn't be saved.

Like instances of dungeons in mmorpgs. And yes i did use workarounds while the system wasn't there in 1.13 but it wasn't as good as fresh dimensions. But i don't want to discuss workarounds in this PR, i can do that in discord if you need.

Reply:
Pretty sure we talked about this on Discord, Your usecase is a bad one. Throwing away entire dimensions is not a good system.